ntbs-service/Services/AzureAdDirectoryService.cs

```diff
@@ -66,10 +66,15 @@ public async Task<IList<Claim>> BuildRoleClaimsForUser(string userPrincipalName)
         {
             var roleClaims = new List<Claim>();
 
+            // If the username contains an apostrophe then we need to escape it by using a double apostrophe in
+            // the filter clause. I don't think other characters would cause an issue, as they will be URL encoded
+            // by the Graph API client.
+            var escapedUserName = userPrincipalName.Replace("'", "''");
```

Comment:
Since this is only used in one place, I decided that it wasn't worth creating an extension method for it.